crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs

```diff
@@ -80,6 +78,7 @@ impl AsyncToGenerator {
         }
 
         *stmts = stmts_updated;
+        stmts.visit_mut_children_with(self);
```

Comment:
This is the biggest part of the headache. I know this is not feasible, as this'll require redundant visit against `extra_stmt` generated. But the order of visit actually matters, if this occurs before visiting individual stmts, inner scope of fn gets visited first before class_method reaches and there's no easy way to detect if fn's `this` scope is inside of class or not.

Reply:
Wait, why do we need to know whether it's inside class or not?

Reply:
to determine captured `this` to be inserted in the fn. (`var _this = this`), not all of this context needs this.

Reply:
Oh, you mean whether to insert `var _this = this` to parent scope or within function? If I'm understanding `babel/helper-wrap-function` correctly, there's a definite list for that we don't need to access parent scope: `class_method` `class_static_property` and `object_property`, we could label them manually

Reply:
that's interesting. Curious if we have equivalent already? @kdy1 otherwise I guess we may need it as well.

Reply:
It may makes bit tricky to pass renamed ident across maybe, but have to see it if that's the direction we'll go.

Reply:
Yes, 'this_id: a &'a mut Ident' seems to be a bit weird

Reply:
I agree that we need one, [contains_this_expr](https://docs.rs/swc_ecma_utils/0.52.0/swc_ecma_utils/fn.contains_this_expr.html) is a similar one, but it's a visitor.

Reply:
After thinking a lot, I concluded that it's fine because `visit_mut_stmts` of `Actual` will prevent it from being too deep.